src/file-upload/file-list.tsx

```diff
@@ -169,28 +172,30 @@ export const FileList = ({
     ): FileItemRenderModes => {
         if (!fileItems || fileItems.length === 0) return {};
 
-        const updatedRenderModes: FileItemRenderModes = { ...renderModes };
+        const newRenderModes: FileItemRenderModes = {};
 
         for (const item of fileItems) {
+            /**
+             * We retain the cancelled edits to prevent the edit
+             * display from re-rendering
+             */
             if (
-                !updatedRenderModes[item.id] ||
-                updatedRenderModes[item.id] !== "cancelled-edit"
+                renderModes[item.id] &&
+                renderModes[item.id] === "cancelled-edit"
             ) {
-                /**
-                 * If new item or if previously did not have edit cancelled,
-                 * we will just update the render mode according to the
-                 * description value.
-                 *
-                 * If editing was cancelled previously, we will leave it as
-                 * it is and not render the edit display
-                 */
-                updatedRenderModes[item.id] = shouldRenderEditMode(item)
+                newRenderModes[item.id] = renderModes[item.id];
+            } else if (item.errorMessage) {
+                newRenderModes[item.id] = "error";
+            } else if (item.progress && item.progress < 1) {
```

Comment:
progress = 0 is also falsy, can check for undefined/null or `item.progress === "number"` instead

Reply:
For this, actually i am just intending to check if progress is specified and it is less than 1. I would think 
```
item.progess < 1 
```
would suffice since `progress === undefined` will resolve to false